README.md

```diff
@@ -1,205 +1,278 @@
+
+Roxygen2 is a package you will need to install in R. Roxygen2 will create .Rd files for every function you document in the _/man_ folder, as well as a DESCRIPTION and NAMESPACE file.
+
+Refer to our [R documentation checklist](RDocChecklist.md) each time you are plan to commit changes with alterations to C documentation.
+
+#### Unit Testing <a name="rtest"/>
+
+We use the [testthat](https://journal.r-project.org/archive/2011-1/RJournal_2011-1_Wickham.pdf) package for unit testing in R. Within each of our R package repositories there is a _tests/testthat_ directory. Within the _tests_ folder there is testthat.R, which guides R to to test all files in the _testthat_ folder during the R CMD check. Each file within the _/testthat_ folder should begin with __test__ and each of these files contains multiple related tests. Typically, we have at least a test file for each corresponding file in the _/R_ folder, but it is possible that there might need to be many test files for each file in the _/R_ folder.
+
+Each test should be designed to test the expectation or multiple expectations of a function. Is the output the right value or the right class? Are the values equal to a reference value? Is there an error message, warning, or message when we want there to be? There is more information on the variety of available test functions and how to use theme [here.](http://r-pkgs.had.co.nz/tests.html#test-tests)
+
+Each and every function should have a test. Our goal is 100% code coverage. As you write tests, check that they are working by with Ctrl/Cmd-Shift-T or devtools::tests().
+
+### C Code Standards <a name="cinfo"/>
+
+#### Style Guide <a name="cstyle"/>
+
+#### Code Development <a name="cdevel"/>
+
+#### Documentation  <a name="cdoc"/>
+
+We use Doxygen to write documentation in C. The style guide and rules for Doxygen can be found [here](http://www.stack.nl/~dimitri/doxygen/).
+
+To create a document from your Doxygen code, navigate to the desired directory (i.e. 'Git/SOILWAT2'), and then simply execute the 'Doxygen' command in the terminal along with the configuration file: `Doxygen Doxyfile`
+This will create a index.html file in /doc folder that you can open in any web browser.
+
+Refer to our [C documentation checklist](CDocChecklist.md) each time you are plan to commit changes with alterations to C documentation.
+
+### Unit Testing <a name="ctest"/>
+
+We use [googletest](https://github.com/google/googletest/blob/master/googletest/docs/primer.md) for unit testing in C. Within our C respoistories there is a /test folder and a /googletest folder. The /googletest folder is a submodule link to the googletest functionality and should not be editted. In the /test folder, for each .c file there is a corresponding test_.cc file which contains the unit tests for the functions found in the .c file. For example, for the SOILWAT2/SW_Site.c file there is a SOILWAT2/test/test_SW_Site.cc file for unit tests.
```

Comment:
Maybe expand that `googletest` is a C++ framework and our code is C. Thus, there are some quirks that are not normal, e.g., 
- that we need a separate compiler for the unit tests
- that we cannot currently include `SW_Output.c` but instead have `SW_Output_mock.c` for the unit tests
- that we need to pay attention to the global states and reset them with a call to `Reset_SOILWAT2_after_UnitTest();`
- that we cannot use several aspects of the `googletest` framework